### ISSUE TITLE
Drupal Code Generator generates Constraint Validator with wrong constructor definition

### ISSUE THREAD

**Issue by bhanu951** [Chi-teck/drupal-code-generator#118]
## Description
Drupal Code Generator generates Constraint Validator with wrong constructor definition.

```
ArgumentCountError: Too few arguments to function Drupal\test_entity\Plugin\Validation\Constraint\TestConstraintValidator::__construct(), 0 passed in /var/www/html/web/core/lib/Drupal/Core/DependencyInjection/ClassResolver.php on line 31 and exactly 5 expected in Drupal\test_entity\Plugin\Validation\Constraint\TestConstraintValidator->__construct() (line 20 of modules/custom/test_entity/src/Plugin/Validation/Constraint/TestConstraintValidator.php).
Drupal\Core\DependencyInjection\ClassResolver->getInstanceFromDefinition('Drupal\test_entity\Plugin\Validation\Constraint\TestConstraintValidator') (Line: 44)
Drupal\Core\Validation\ConstraintValidatorFactory->getInstance(Object) (Line: 200)
Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateConstraints(Object, '00000000000006810000000000000000', Array) (Line: 154)
Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateNode(Object) (Line: 164)
Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validateNode(Object, Array, 1) (Line: 106)
Drupal\Core\TypedData\Validation\RecursiveContextualValidator->validate(Object, NULL, NULL) (Line: 93)
Drupal\Core\TypedData\Validation\RecursiveValidator->validate(Object) (Line: 132)
Drupal\Core\TypedData\TypedData->validate() (Line: 485)
Drupal\Core\Entity\ContentEntityBase->validate() (Line: 188)
Drupal\Core\Entity\ContentEntityForm->validateForm(Array, Object)
call_user_func_array(Array, Array) (Line: 82)
Drupal\Core\Form\FormValidator->executeValidateHandlers(Array, Object) (Line: 275)
Drupal\Core\Form\FormValidator->doValidateForm(Array, Object, 'test_entity_add_form') (Line: 118)
Drupal\Core\Form\FormValidator->validateForm('test_entity_add_form', Array, Object) (Line: 593)
Drupal\Core\Form\FormBuilder->processForm('test_entity_add_form', Array, Object) (Line: 325)
Drupal\Core\Form\FormBuilder->buildForm(Object, Object) (Line: 73)
Drupal\Core\Controller\FormController->getContentResult(Object, Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 592)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 121)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 182)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 76)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 44)
Drupal\redirect_after_login\RedirectMiddleware->handle(Object, 1, 1) (Line: 58)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\StackedHttpKernel->handle(Object, 1, 1) (Line: 704)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

## Steps to Reproduce
```drush gen plugin:constraint

 Welcome to constraint generator!
––––––––––––––––––––––––––––––––––

 Module machine name:
 ➤ test_entity

 Module name [Test entity]:
 ➤

 Plugin label:
 ➤ Test Constraint

 Plugin ID [TestEntityTestConstraint]:
 ➤

 Plugin class [TestConstraintConstraint]:
 ➤ TestConstraint

 Type of data to validate:
  [1] Raw value
  [2] Item
  [3] Item list
  [4] Entity
 ➤ 1

 Would you like to inject dependencies? [No]:
 ➤ yes

 Type the service name or use arrows up/down. Press enter to continue:
 ➤ entity_type.manager

 Type the service name or use arrows up/down. Press enter to continue:
 ➤ database

 Type the service name or use arrows up/down. Press enter to continue:
 ➤

 The following directories and files have been created or updated:
–––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
 • /var/www/html/web/modules/custom/test_entity/src/Plugin/Validation/Constraint/TestConstraint.php
 • /var/www/html/web/modules/custom/test_entity/src/Plugin/Validation/Constraint/TestConstraintValidator.php

```
Now add the generated Constraint Validator to any field , for example title field in node entity.

**test_entity.module file**

```
use Drupal\Core\Entity\EntityTypeInterface;

/**
 * Implements hook_entity_bundle_field_info_alter().
 */
function test_entity_entity_bundle_field_info_alter(&$fields, EntityTypeInterface $entity_type, $bundle) {

  if ($entity_type->id() === 'node' && isset($fields['title'])) {

    // Add our custom validation to the title field of node entity.
    $fields['title']->addConstraint('TestEntityTestConstraint');
  }

}

```

## Actual Behavior
Doesnt generate a working Constraint Validator Pluging.

## Expected Behavior
Should generate a working Constraint Validator.

## Version Info
```
composer show chi-teck/drupal-code-generator
name     : chi-teck/drupal-code-generator
descrip. : Drupal code generator
keywords :
versions : * 3.2.0
type     : library
license  : GNU General Public License v2.0 or later (GPL-2.0-or-later) (OSI approved) https://spdx.org/licenses/GPL-2.0-or-later.html#licenseText
homepage :
source   : [git] https://github.com/Chi-teck/drupal-code-generator.git 6883a423f21446b16f8b9a2d90cc5abd4bc3e5d6
dist     : [zip] https://api.github.com/repos/Chi-teck/drupal-code-generator/zipball/6883a423f21446b16f8b9a2d90cc5abd4bc3e5d6 6883a423f21446b16f8b9a2d90cc5abd4bc3e5d6
path     : /var/www/html/vendor/chi-teck/drupal-code-generator
names    : chi-teck/drupal-code-generator

support
issues : https://github.com/Chi-teck/drupal-code-generator/issues
source : https://github.com/Chi-teck/drupal-code-generator/tree/3.2.0

autoload
psr-4
DrupalCodeGenerator\ => src

requires
ext-json *
php >=8.1.0
psr/event-dispatcher ^1.0
psr/log ^3.0
symfony/console ^6.3
symfony/dependency-injection ^6.3.2
symfony/filesystem ^6.3
symfony/string ^6.3
twig/twig ^3.4

requires (dev)
chi-teck/drupal-coder-extension ^2.0.0-alpha4
drupal/coder 8.3.20
drupal/core 10.1.x-dev
ext-simplexml *
phpspec/prophecy-phpunit ^2.0
phpunit/phpunit ^9.5
squizlabs/php_codesniffer ^3.7
symfony/var-dumper ^6.3
symfony/yaml ^6.3
vimeo/psalm ^5.14.0

conflicts
squizlabs/php_codesniffer <3.6
```
**Drush Version: 12.2.0**

## Suggested Change
Constraint Validator Plugin implements `ContainerFactoryPluginInterface` where as it should implement `ContainerInjectionInterface`.

Working Constraint Validator from Core [TaxonomyTermHierarchyConstraintValidator](https://git.drupalcode.org/project/drupal/-/blob/11.x/core/modules/taxonomy/src/Plugin/Validation/Constraint/TaxonomyTermHierarchyConstraintValidator.php)

**Comment by Chi-teck** [comment]
This has already been [fixed](https://github.com/Chi-teck/drupal-code-generator/commit/9a8a9cbee122c13c1c645321b3cefc9591b33af2) in 3.3.0.  Can you update your DCG version and confirm it works as expected?

**Comment by bhanu951** [comment]
Hi @Chi-teck Thanks for the response. Yes indeed it got fixed in 3.3.0.
